app/services/foreman_ansible/variables_importer.rb

```diff
@@ -48,22 +48,21 @@ def import_new_role(role_name, new_roles)
     def initialize_variables(variables, role)
       variables.map do |variable_name, variable_default|
         variable = AnsibleVariable.find_or_initialize_by(
-          :key => variable_name
+          :key => variable_name,
+          :ansible_role_id => role.id
         )
-        if variable.new_record?
-          variable.assign_attributes(:default_value => variable_default,
-                                     :key_type => infer_key_type(variable_default),
-                                     :imported => true)
-        end
-        variable.ansible_role = role
+        variable.assign_attributes(:default_value => variable_default,
```

Comment:
@xprazak2 notice this line. Previously, the assignment was happening only `if new_record?` now it's happening every time, so it solves [#26485](https://projects.theforeman.org/issues/26485) as it reimports `default_value`

Reply:
(then the change assesment has totally different meaning obviously, as it asses change of value instead of change of `role_id`)

Reply:
[#26485](https://projects.theforeman.org/issues/26485) describes a selective import based whether the default value has been changed in foreman or not, that is why I was confused and do not think this change actually implements the behaviour that is described there. 

I am willing to merge as is, because it fixes the import problems for vars with the same name.

Reply:
It does solve the issue I believe. This line imports to existing variable and https://github.com/theforeman/foreman_ansible/pull/335/files#diff-e2db0db7ab2083dffdd6509a1e79702fR64 throw away the unchanged ones. So it imports all variables, but doesn't show up in importer if there is no change. That is exactly what the issue is suggesting, right?

Reply:
Oh, I see it now, thanks for bearing with me.